### PR TITLE
Make use of ES2015 features

### PR DESCRIPTION
**Pull request by realityking**
Nothing major but `removeDups` got eliminated.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,24 +24,24 @@ var Config = require('./lib/Config');
...
   dedupeNonExistent(config.nonExistent);
   debug('deduped list of nonExistent partials: ', config.nonExistent);
 
-  var tree;
+  let tree;
   if (config.isListForm) {
     debug('list form of results requested');
 
-    tree = removeDups(results);
+    tree = Array.from(new Set(results));
```
Comment (mrjoelkemp) [inline index.js]: How does the runtime and space complexity compare to the old implementation?

Comment (realityking) [reply index.js]: I used the run time of the tests suite as a quick sanity check while I worked on this and it didn't change significantly.

According to https://medium.com/@jakubsynowiec/unique-array-values-in-javascript-7c932682766c it should be quite a bit faster than the old code.

Comment (realityking) [reply index.js]: That’s fair. I’ll back that part out once I get to my notebook.

**Comment by mrjoelkemp** [comment]
Thanks for this. I'd be happy to merge the const/let changes, use of `use strict`, and config as a class changes. 

I need to dig deeper into the change that uses Set. I'd be willing to review that in a follow up if you'd like. Sorry for the delay in reviewing; trying to carve out time for open source stuff.

**Comment by realityking** [comment]
No worries about the delay - it's not like these are bug fixes or otherwise blockers.

I removed the commit removing `removeDups` but I did add one that uses a `Set` internally in `removeDups`. I hope that's ok.

**Comment by mrjoelkemp** [comment]
Thanks @realityking! I pushed this in a patch release.

**Pull request merged**